cmd/orch-tester/orch_tester.go

```diff
@@ -0,0 +1,558 @@
+	media := flag.String("media", bcastMediaPort, "Broadcaster HTTP port")
+	rtmp := flag.String("rtmp", bcastRTMPPort, "broadcaster RTMP port")
+	leaderboard := flag.String("leaderboard", "127.0.0.1:3001", "HTTP Address of the serverless leadearboard API")
+	leaderboardSecret := flag.String("leaderboard-secret", "", "Secret for the Leaderboard API")
+	subgraph := flag.String("subgraph", "https://api.thegraph.com/subgraphs/name/livepeer/livepeer-canary", "Livepeer subgraph URL")
+	// Video config
+	videoFile := flag.String("video", "official_test_source_2s_keys_24pfs_30s.mp4", "video file to use, has to be present in stream-tester root")
+	numProfiles := flag.Int("profiles", 3, "number of video profiles to use on the broadcaster")
+	repeat := flag.Int("repeat", 1, "number of times to repeat the stream")
+	simultaneous := flag.Int("simultaneous", 1, "number of times to run the stream simultaneously")
+	flag.Parse()
+
+	if *region == "" {
+		log.Fatal("region is required")
+	}
+
+	metricsURL, err := defaultAddr(*metrics, "127.0.0.1", prometheusPort)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	streamTesterURL, err := defaultAddr(*streamTester, "127.0.0.1", streamTesterPort)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	leaderboardURL, err := defaultAddr(*leaderboard, "127.0.0.1", "3001")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	subgraphURL, err := defaultAddr(*subgraph, "127.0.0.1", "8080")
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	rtmpUint, err := strconv.ParseUint(*rtmp, 10, 16)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	mediaUint, err := strconv.ParseUint(*media, 10, 16)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	streamer, err := newStreamerClient(streamTesterURL, metricsURL, leaderboardURL, *leaderboardSecret, subgraphURL)
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	orchestrators, err := streamer.getOrchestrators()
+	if err != nil {
+		log.Fatal(err)
+	}
+
+	refreshWait := 90 * time.Second
+
+	for _, o := range orchestrators {
+		time.Sleep(refreshWait)
+
+		req := &streamerModel.StartStreamsReq{
+			Host:            *broadcaster,
+			RTMP:            uint16(rtmpUint),
+			Media:           uint16(mediaUint),
+			Repeat:          uint(*repeat),
+			Simultaneous:    uint(*simultaneous),
+			Orchestrators:   []string{o.ServiceURI},
+			ProfilesNum:     *numProfiles,
+			DoNotClearStats: false,
+			MeasureLatency:  false,
+			HTTPIngest:      true,
+			FileName:        *videoFile,
+		}
+
+		startTime := time.Now().Unix()
+		mid, err := streamer.startStream(req)
+		if err != nil {
+			glog.Error(err)
+			continue
+		}
+
+		glog.Infof("Started stream for orchestrator %v", o.Address)
+
+		apiStats := &apiModels.Stats{
+			Region:       *region,
+			Timestamp:    startTime,
+			Orchestrator: o.Address,
+		}
+
+		// Make sure manifest ID exists before getting stats
+		time.Sleep(15 * time.Second)
+		// wait for stream to finish transcoding
+		streamerStats, err := streamer.getFinishedStats(mid)
+		if err != nil {
+			glog.Error(err)
```

Comment:
Should we continue code execution in this loop iteration if an error is encountered here?

Reply:
fixed in aed8982